ui/components/MesheryPatterns.js

```diff
@@ -799,7 +801,23 @@ function MesheryPatterns({
     });
   };
 
+  const getDesignVersion = (design) => {
+    if (design.visibility === 'published') {
+      return design.catalog_data.published_version;
+    } else {
+      try {
+        const parsedYaml = yaml.load(design.pattern_file);
+        return parsedYaml.version;
+      } catch (error) {
+        console.error('Error parsing YAML:', error);
+        return 'Error';
+      }
+    }
+  };
```

Comment:
Please move this code to utils directory

Reply:
Also we don't need to return "Error" from this function, console will work

Reply:
Also Instead of identifying the version in parent component and passing to info modal we can do that in Info modal itself, after you will move this to utils

Reply:
Done